Feature/documents module: refactor and frontend updates

SreehariVinayan opened the pull request:
# Pull Request: Documents Module – Hook-Service Architecture, UI Enhancements & Template Management

## Overview

**Related Issue(s):** None

This PR merges previous commits and introduces the latest Template Management page updates.

### Previous Changes
#### What (has changed)
- Introduced **Hook-Service architecture** for document-related pages  
- Added **DocumentsTable** with actions and utilities  
- Created **Dashboard and Generate Document Page UI** based on Figma design  
- Reorganized utilities, including renaming and moving files  
- Deleted outdated `util/logic` structure and replaced with reusable hooks and services  

#### Why (was the change introduced)
- To standardize frontend architecture across modules  
- Improve maintainability, testability, and error handling  
- Provide a scalable foundation for API integration and document generation  

#### How (the change was implemented)
- Services created for API interactions, placed in `/services`  
- Custom hooks added to encapsulate state management (loading, error)  
- `page.tsx` updated to consume hooks/services  
- Utility created to extract error messages from API responses  
- Legacy files under `util/logic` removed (**breaking change**)  

---

### Latest Changes
#### What (has changed)
- Added a **Template Management** page UI inside the `documents` module.  
- Included a **placeholder** for the future **WYSIWYG editor** (as specified in the SRS) with Preview, Save, and Copy button functionalities still pending implementation.  
- **Restructured** the `documents` module by deleting legacy hooks/services and creating a clearer folder structure for:
  - `dashboard/`
  - `generate-document/`
  - `template-management/`  
- Updated related components (`DocumentsTable`, pages, utils) to use the new paths.
- Added Document Permissions & Role Assignment Page UI:
  - Components: DocumentPermissionsBox, UserRoleAssignmentBox
  - Hooks: useFetchPermissions, useFetchRoles, useFetchUsersAndGroups, useRoleAssignment, useSavePermissions
  - Services for fetching/saving permissions, roles, users, and updating role assignments

#### Why (was the change introduced)
- To support upcoming template creation/editing features.
- To remove outdated logic and improve maintainability by separating concerns into dedicated folders.
- To support upcoming permissions management and role assignments

#### How (the change was implemented)
- Created a new `template-management` page and basic UI.
- Added placeholder elements for the WYSIWYG editor and future action buttons.
- Deleted old hooks and services, then updated all imports to the new structure.
- Created new components/hooks/services for managing permissions and role assignments
- Updated existing pages/hooks to consume new components and hooks
- Migrated interfaces to types and updated all affected imports
---

### Type of Change
- [x] New Feature  
- [x] Refactor (breaking structure changes)  

### Domain of Change
- [x] Frontend/ UI  

---

## Implementation Strategy
The new Template Management page acts as a container for the upcoming WYSIWYG editor.  
All document-related logic has been organized into three focused folders to reduce coupling and make future additions easier.  
The hook-service pattern introduced earlier continues to provide consistent state and API management.

---

## How Has This Been Tested?
- Verified local build compiles with no errors or warnings.
- Navigated to all updated pages (DocumentsTable, Generate Document, Template Management) and confirmed UI renders without runtime issues.
- Checked error handling flows with dummy data injected on error.

---

## Screenshots/ Visuals (Frontend/ UI)
<img width="1738" height="942" alt="image" src="https://github.com/user-attachments/assets/5871ce17-ea78-4f5a-b0a0-a7ee0ca8f633" />

<img width="1738" height="942" alt="image" src="https://github.com/user-attachments/assets/9138484c-ae7e-4543-8cb1-3d9f5f4f1b5d" />


<img width="1738" height="942" alt="image" src="https://github.com/user-attachments/assets/391fb3a8-08d5-4073-9a5a-e7c6a3a16ea0" />

<img width="1722" height="811" alt="image" src="https://github.com/user-attachments/assets/de905f01-1c0e-4c6c-8423-f54c92b69e7c" />


---

## Developer Checklist
- [x] Self-review completed (no unused code or typos).
- [x] Code compiles/builds without errors or warnings.
- [ ] Tests added/passing *(if applicable)*.
- [x] Documentation updated where necessary.
- [x] No sensitive information exposed.
- [x] All dependent changes merged.

---

## Other Information
**BREAKING CHANGE:**  
- Old `util/logic` structure removed. All imports must be updated to the new hook-service structure.  
- All old hooks and services have been deleted. Consumers must update imports to the new `dashboard/`, `generate-document/`, and `template-management/` paths.
- Interfaces replaced by types; all old interface imports must be updated

SreehariVinayan committed:
feat(documents): add DocumentsTable with actions and dummy data utilities

SreehariVinayan committed:
feat(documents): add Generate Document Page UI and reorganize utility files

- Added basic UI for Generate Document Page based on Figma
- Renamed `download.ts` → `handleDocumentDownload.ts` and moved it to `utils/logic/`
- Created `utils/logic/` folder and moved placeholder logic files there for future use
- Updated references accordingly

BREAKING CHANGE: imports of any moved/renamed files (e.g., `download.ts`) must be updated to their new paths in `utils/logic/`

SreehariVinayan committed:
Merge branch 'frontend' into feature/documents

SreehariVinayan committed:
Merge branch 'frontend' into feature/documents

SreehariVinayan committed:
refactor(documents): Restructure code for hooks-service format in dashboard and document generation pages

- Created services which calls the apis required for the documents pages and put it in services folder
- Created new hooks which calls the services and manages the common states like loading and error
- Updated the page.tsx of each page to adapt to the new Hook-Service structure
- Deleted the util/logic folder which contained the old strcuture
- Created a util file for getting the error message extracted from the api error which is reused in all hooks

BREAKING CHANGE: removed old util/logic files, must update imports and adapt to the new hook-service structure

SreehariVinayan committed:
feat(documents): add template management page with WYSIWYG editor placeholder and restructure module

Created UI for Template Management page.
Added a placeholder for the future WYSIWYG editor as specified in the SRS,
with preview, save, and copy button functionalities still pending implementation.

Refactored the documents module by removing legacy hooks and services and creating
a clearer folder structure for dashboard, generate-document, and template-management
logic. Updated related components (DocumentsTable, pages, utils) to align with the
new structure.

BREAKING CHANGE: All old hooks and services have been deleted and their imports must
be updated to the new dashboard/, generate-document/, and template-management/ paths.

JVP555 reviewed (dismissed):
1. Move the currently defined components from apps/client-web/src/app/modules/documents to features/documents.

2. Refactor the components to use type-based structures consistently.

3. Review and ensure all API logic is correctly implemented and resides in services/hooks, not in the components.

SreehariVinayan committed:
feat(documents): created access control page

Added components, hooks, and services for managing document permissions and role assignments:
- DocumentPermissionsBox, UserRoleAssignmentBox
- useFetchPermissions, useFetchRoles, useFetchUsersAndGroups, useRoleAssignment, useSavePermissions
- Services for fetching and saving permissions/roles/users, and updating role assignments

Updated existing pages and hooks to better structure:
- Modified page.tsx, dashboard/useFetchDocuments.ts, and template-management/page.tsx

SreehariVinayan committed:
refactor(documents): convert interfaces to types and restructure types folder

- Moved all interfaces to a dedicated types folder, with separate files per page.
- Updated imports across the module accordingly.
- Corrected type errors and unsafe member accesses flagged by ESLint.
- Performed minor refactors and bug fixes in components and hooks.

BREAKING CHANGE: Interfaces are replaced by types in the documents module.
All imports from old interface locations must be updated.

JVP555 committed:
Merge branch 'frontend' into feature/documents
